components/board/genericlinux/hw_pwm.go

```diff
@@ -0,0 +1,174 @@
+// Package genericlinux is for Linux boards. This particular file is for using sysfs to
+// interact with PWM devices. All of these functions are idempotent: you can double-export a pin or
+// double-close it with no problems.
+package genericlinux
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"sync"
+
+	"github.com/pkg/errors"
+)
+
+type pwmDevice struct {
+	// These values are immutable
+	chipPath string
+	line     int
```

Comment:
Is there anything enforcing immutability here?

Reply:
Alas, I don't think Go has that capability. :disappointed: It's just a handshake agreement, which is why the comment saying what's happening is important.